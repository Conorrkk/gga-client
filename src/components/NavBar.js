import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import { Link, useNavigate } from "react-router-dom";
import { logout } from "../api";

function NavBar() {

  const navigate = useNavigate();

  const handleLogout = async () => {
    try {
      await logout()
      navigate("/")
    } catch (error) {
      console.error("Logout error:", error)
    }
  }

  return (
    <div>
      <Navbar bg="dark" className="custom-navbar" variant={"dark"} expand="lg">
        <Navbar.Brand className="ms-2" href="#">
          GGAnalytics
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="navbarScroll" />
        <Navbar.Collapse id="navbarScroll">
          <Nav
            className="mr-auto my-2 my-lg-0"
            style={{ maxHeight: "100px" }}
            navbarScroll
          >
            <Nav.Link as={Link} to="/dashboard">
              Dashboard
            </Nav.Link>
            <Nav.Link as={Link} to="/teams">
              Teams
            </Nav.Link>
            <Nav.Link as={Link} to="/createMatch">
              Create Match
            </Nav.Link>
            <Nav.Link as={Link} to="/matchHistory">
              Match History
            </Nav.Link>
            <Nav.Link onClick={handleLogout}>
              Logout
            </Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    </div>
  );
}

export default NavBar;
