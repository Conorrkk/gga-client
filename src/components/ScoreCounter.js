import React, { useState } from "react";
import Button from "react-bootstrap/Button";
import Card from "react-bootstrap/Card";

import Row from "react-bootstrap/Row";

function ScoreCounter({ onUpdateGoalAgainst, onUpdatePointAgainst }) {
  // state to score goals
  const [goals, setGoals] = useState(0);
  // state to score points
  const [points, setPoints] = useState(0);

  // if goals are
  const incrementGoals = () => {
    if (goals < 50) {
      setGoals(goals + 1);
      onUpdateGoalAgainst(goals + 1);
    }
  };

  const decrementGoals = () => {
    if (goals > 0) {
      setGoals(goals - 1);
      onUpdateGoalAgainst(goals - 1);
    }
  };

  const incrementPoints = () => {
    if (points < 50) {
      setPoints(points + 1);
      onUpdatePointAgainst(points + 1);
    }
  };

  const decrementPoints = () => {
    if (points > 0) {
      setPoints(points - 1);
      onUpdatePointAgainst(points - 1);
    }
  };

  return (
    <Card className="scoreline-against">
      <Card.Body>
        <Row className="mb-2">
          <div className="score-section">
            <div className="score-actions">
              <Button className="scoreline-button" variant="outline-dark me-1" onClick={incrementGoals}>
                {goals}
              </Button>

              <Button className="scoreline-button" variant="outline-dark" onClick={incrementPoints}>
                {points}
              </Button>
            </div>
          </div>
        </Row>
        <Row>
          <div className="score-section">
            <div className="score-actions">
              <Button className="scoreline-button" variant="outline-warning me-2" onClick={decrementGoals}>
                -
              </Button>
              <Button className="scoreline-button" variant="outline-warning" onClick={decrementPoints}>
                -
              </Button>
            </div>
          </div>
        </Row>
      </Card.Body>
    </Card>
  );
}

export default ScoreCounter;
