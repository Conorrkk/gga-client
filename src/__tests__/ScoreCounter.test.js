import React from 'react';
import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
import NavBar from "../components/NavBar"
import ScoreCounter from '../components/ScoreCounter';


it('renders correctly', () => {
    const tree = renderer
    .create(<ScoreCounter/>)
    .toJSON();

    expect(tree).toMatchSnapshot();
});