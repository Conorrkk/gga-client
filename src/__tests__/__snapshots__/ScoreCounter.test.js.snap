// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="scoreline-against card"
>
  <div
    className="card-body"
  >
    <div
      className="mb-2 row"
    >
      <div
        className="score-section"
      >
        <div
          className="score-actions"
        >
          <button
            className="scoreline-button btn btn-outline-dark me-1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="scoreline-button btn btn-outline-dark"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            0
          </button>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="score-section"
      >
        <div
          className="score-actions"
        >
          <button
            className="scoreline-button btn btn-outline-danger me-2"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            -
          </button>
          <button
            className="scoreline-button btn btn-outline-danger"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
